packages/data/src/components/with-select/with-select-new.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * WordPress dependencies
+ */
+import { createHigherOrderComponent, pure } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import useSelect from '../use-select';
+
+const withSelect = ( mapSelectToProps ) => createHigherOrderComponent(
+	( WrappedComponent ) => pure(
```

Comment:
Can I ask why 'pure' is necessary here? can't we just pass the `Object.values( ownProps }`  as a second argument for `useSelect` instead?

Reply:
We tried, but `deps` expects the number of items passed to be constant and `ownProps` gets items removed/added.

Reply:
Mmm makes sense.